packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js

```diff
@@ -54,27 +59,60 @@ export default ({ coreSagas }) => {
     }
   }
 
-  const sell = function * (payload) {
+  const sell = function * () {
     try {
-      const sellTrade = yield call(coreSagas.data.coinify.sell, payload)
+      const trade = yield call(coreSagas.data.coinify.sell)
+      yield put(A.coinifyLoading())
 
-      if (!sellTrade) {
+      if (!trade) {
         const trade = yield select(selectors.core.data.coinify.getTrade)
         const parsed = JSON.parse(trade.error)
 
         yield put(A.coinifyFailure(parsed))
         return
       }
+      const p = yield select(sendBtcSelectors.getPayment)
+      let payment = yield coreSagas.payment.btc.create(
+        { payment: p.getOrElse({}), network: settings.NETWORK })
+      payment = yield payment.amount(parseInt(trade.sendAmount))
 
-      yield put(A.coinifyNotAsked())
-      yield put(A.coinifyNextCheckoutStep('isx'))
+      // QA Tool: manually set a "to" address on the payment object for testing sell
+      const qaState = yield select()
+      const qaAddress = path(['qa', 'qaSellAddress'], qaState)
+
+      if (qaAddress) {
+        payment = yield payment.to(qaAddress)
+      } else {
+        payment = yield payment.to(trade.receiveAddress)
+      }
+      payment = yield payment.to(trade.receiveAddress)
```

Comment:
Would this overwrite the `if(qaAddress)` block

Reply:
Indeed, that's just a line I forgot to remove, good catch!